aws/database/rds/vpc.tf

```diff
@@ -0,0 +1,47 @@
+locals {
+  vpc_cidr_block = "10.0.0.0/24"
+}
+
+resource "aws_vpc" "main" {
+  cidr_block           = local.vpc_cidr_block
+  enable_dns_support   = var.publicly_accessible # if DB instance is publicly accessible must be enabled
+  enable_dns_hostnames = var.publicly_accessible # if DB instance is publicly accessible must be enabled
+}
+
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+resource "aws_subnet" "instances_subnets" {
+  count             = 2
+  vpc_id            = aws_vpc.main.id
+  cidr_block        = cidrsubnet(local.vpc_cidr_block, 4, count.index)
+  availability_zone = data.aws_availability_zones.available.names[count.index]
+
+  tags = {
+    Name = "${var.db_name}-subnet-${count.index}"
+  }
+}
+
+resource "aws_db_subnet_group" "main" {
+  subnet_ids = [aws_subnet.instances_subnets.0.id, aws_subnet.instances_subnets.1.id]
+}
+
+resource "aws_security_group" "main" {
+  name   = "db_traffic"
+  vpc_id = aws_vpc.main.id
+}
+
+# Only allow incoming traffic from all ipv4 ips on port 5432 if publicly accessible
+# if used in production this SHOULD be limited further.
+# If this is set to false, the expectation is that accessing party's
+# should add a rule to the vpc that they are allowed to access
```

Comment:
Currently there is no configuration for setting this up when setting publicly available to false.
It also looks like it is a little bit more effort and we could/should postpone this to another PR.

We could perhaps use vpc-perring and security-group-referencing https://docs.aws.amazon.com/vpc/latest/userguide/security-group-rules.html#security-group-referencing for the case that we want to have it private and only allow access from another asg or ec2 instance. Did just quickly skim through, there might be better solutions.

What do you think about removing the 'publicly_accessible' option from this module as long as we don't have the feature for 'privately_accessible' built out? Otherwise it could be misleading. Or rather, we will create instances where we cannot connect to (without adding stuff manually) which should (imo) not be the usage for our modules.

Reply:
@thiemogries & @xddq let's have a quick 5 min chat about this after today's daily?

Reply:
I think having the option to make it not public is already useful, even if it means having to manually create a vpc peering and/or setting up security groups. This way this module could be used with some extra work. Otherwise it couldn't be used at all.

---

Also in high likelihood this will be implemented very soon anyway and having to refactor this out only to reimplement it again in a few days isn't worth the hassle. I would keep it the way it is and add the ease-of-use-private-accessible feature in another pr.

Reply:
I created a ticket: https://app.zenhub.com/workspaces/they-organisation-dev-overview-61e82d3d91e1df001c4d3ca3/issues/gh/they-consulting/they-terraform/64